packages/valory/skills/agent_performance_summary_abci/behaviours.py

```diff
@@ -268,6 +268,17 @@ def _save_agent_performance_summary(
 
     def async_act(self) -> Generator:
         """Do the action."""
+        if not self.params.is_agent_performance_summary_enabled:
+            self.context.logger.info(
+                "Agent performance summary is disabled. Skipping fetch and save."
+            )
+            payload = FetchPerformanceDataPayload(
+                sender=self.context.agent_address,
+                vote=False,
+            )
+            yield from self.finish_behaviour(payload)
+            return
```

Comment:
I believe there is no need to return, because of the `self.set_done()` in the `finish_behaviour`.